openedx/core/djangoapps/credentials_service/models.py

```diff
@@ -1,3 +1,270 @@
 """
 Models for the credentials service.
 """
+import os
+
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+from model_utils.models import TimeStampedModel
+from xmodule_django.models import CourseKeyField
+
+
+MODE_CHOICES = (
+    (_('Honor'), 'honor'),
+    (_('Verified'), 'verified'),
+    (_('Professional'), 'professional'),
+)
+
+
+class AbstractCredential(TimeStampedModel):
+    """
+    Abstract Credentials configuration model to support multitenancy.
+    """
+    PROGRAMS, COURSES = (
+        "programs", "courses",
+    )
+    TYPES_CHOICES = (
+        (PROGRAMS, _("programs")),
+        (COURSES, _("courses")),
+    )
+
+    credential_type = models.CharField(max_length=32, choices=TYPES_CHOICES, default=COURSES)
```

Comment:
This isn't necessary. The credential type is determined by the class (e.g. `CourseCertificate`).